js/privatebin.js

```diff
@@ -4245,27 +4245,34 @@ jQuery.PrivateBin = (function($, RawDeflate) {
                 }
             }
 
-            if (paste.hasOwnProperty('attachment') && paste.hasOwnProperty('attachmentname')) {
-                // version 1 paste
-                Promise.all([
-                    CryptTool.decipher(key, password, paste.attachment),
-                    CryptTool.decipher(key, password, paste.attachmentname)
-                ]).then((attachment) => {
-                    AttachmentViewer.setAttachment(attachment[0], attachment[1]);
-                });
-                PasteViewer.setFormat(paste.meta.formatter);
-                PasteViewer.setText(pastePlain);
-            } else {
+            let format = '',
+                text   = '';
+            if (paste.hasOwnProperty('ct')) {
```

Comment:
Refactor to some function like `getPasteVersion` or maybe better `isVersion2Paste` or so?

Reply:
Sure

Reply:
Given this code smell, I have extracted several of such version specific pieces of logic into generic Paste and Comment classes. The version can now be checked via the `v` attribute that gets set to 1 by default and version 2 pastes will overwrite. See c4b84b2b6bd8